### PR TITLE
Creacion de nuevos servicos institucion cliente y asesor

### PR DESCRIPTION
**Pull request by CamiloArias12**
# Creacion de nuevos servicos institucion y cliente

## Description
Se agregaron los servicios utilizando la arquitectura planteada
1. Asesor (CRUD)
image
image
image
image
image

2.  Integracion con el servico role-cliente 

image
image
4. Actualizar estatus cliente y cargar foto
image
image
5 Cargar logo institucion en un metodo adicional
image

## How Has This Been Tested?
Por medio de hoppscotch se testeo el funcionamiento, 

## Checklist:
- [x]  My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: database/schemas/20240520143101_asesor.up.sql [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-  client_id INT REFERENCES client(client_id) NOT NULL,
-  username VARCHAR(50) NOT NULL,
-  photo VARCHAR(255) NOT NULL,
+  asesor_id INT REFERENCES client(client_id) NOT NULL,
   about VARCHAR(255) NOT NULL,
   created_at TIMESTAMP NOT NULL,
   updated_at TIMESTAMP,
-  deleted_at TIMESTAMP
+  deleted_at TIMESTAMP,
+  PRIMARY KEY(asesor_id)
```
Comment (JosephSC0121) [inline database/schemas/20240520143101_asesor.up.sql:7]: ventajas?

**Review by JosephSC0121 — APPROVED** [review]

File: internal/service/mailing/templates/template-password.mjml
Comment (Camilo-Camargo) [inline internal/service/mailing/templates/template-password.mjml:1]: los mjml no se pueden eliminar

File: internal/service/mailing/templates/template-reset-password.mjml
Comment (Camilo-Camargo) [inline internal/service/mailing/templates/template-reset-password.mjml:1]: los mjml no se pueden eliminar

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

File: internal/service/mailing/templates/template-reset-password.mjml
Comment (CamiloArias12) [reply internal/service/mailing/templates/template-reset-password.mjml:1]: [71de48e](https://github.com/OwlByTech/optitech-api/pull/25/commits/71de48e15b50da5b268a6676c8e689edc4363e93)

File: database/schemas/20240520143101_asesor.up.sql (hunk 1 above)
Comment (CamiloArias12) [reply database/schemas/20240520143101_asesor.up.sql:7]: Se ahorra una columna

File: internal/service/mailing/templates/template-password.mjml
Comment (CamiloArias12) [reply internal/service/mailing/templates/template-password.mjml:1]: [71de48e](https://github.com/OwlByTech/optitech-api/pull/25/commits/71de48e15b50da5b268a6676c8e689edc4363e93)

**Review by Camilo-Camargo — APPROVED** [review]

**Pull request merged**